web/frontend/valexa_app/src/components/FileUpload.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+  <div id="file-upload">
+    <div id="drop" 
+      @dragenter="handleDragover" 
+      @dragover="handleDragover"
+      @drop="handleDrop">
+        Drop a spreadsheet file here to see sheet data
+    </div>
+      <input type="file" id="file" 
+        v-on:change="uploadFile($event.target.files)" 
+      /> ... or click here to select a file
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+
+export default {
+  name: 'FileUpload',
+  props: {
+    url: String,
+  },
+  methods:{
+        handleDrop(e) {
+          e.stopPropagation();
+          e.preventDefault();
+          this.uploadFile(e.dataTransfer.files);
+        },
+        handleDragover(e) {
+          e.stopPropagation();
+          e.preventDefault();
+          e.dataTransfer.dropEffect = 'copy';
+        },
+        uploadFile: function(files){
+            this.file = files[0];
+            var vm = this;
+            let formData = new FormData();
+            formData.append("file", this.file);
+            axios({
+                method: "POST",
+                url: vm.url,
+                data: formData,
+                headers: {
+                    "Content-Type": "multipart/form-data"
+                },
+                responseType: 'json'
+            })
+            .then(function(response){
+                vm.$emit("fileupload-success", response);
+            })
+            .catch(function(error){
+                vm.$emit("fileupload-error", error);
+            })
+            .finally(function(){
+                event.target.value = "";
+                //event.target.type = "file";
+            })
+        }
+    }
+}
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
```

Comment:
This comment is unneeded since with a basic knowledge of Vue.js we know the purpose of the `scoped` keyword.